internal/broker/broker.go

```diff
@@ -502,22 +586,32 @@ func (b *Broker) handleIsAuthenticated(ctx context.Context, session *sessionInfo
 
 		var ok bool
 		// This mode must always come after a authentication mode, so it has to have an auth_info.
-		authInfo, ok = session.authInfo["auth_info"].(authCachedInfo)
+		authInfo, ok = session.authInfo["auth_info"].(tokenPkg.AuthCachedInfo)
 		if !ok {
 			slog.Error("could not get required information")
 			return AuthDenied, errorMessage{Message: "could not get required information"}
 		}
+
+		err = password.HashAndStorePassword(challenge, session.passwordPath)
+		if err != nil {
+			slog.Error(err.Error())
+			return AuthDenied, errorMessage{Message: "could not store password"}
+		}
```

Comment:
Yeah I'll probably keep forgetting that for some time, because I'm used to the opposite (try to avoid doing value assignment and comparison on the same line, because it makes the code a bit less readable, especially when the assignment part is quite long, pushing the comparison further to the right). I tried to get my IDE to warn me when I'm not doing it, and also tried enabling all the linters in golangci-lint, but none supported warnings about this, which to me is an indication that it's not widespread to strictly use that format. I looked if https://go.dev/doc/effective_go has recommendations on it and it does not, but it does put the `if err != nil` on a separate line in the [web server example code](https://go.dev/doc/effective_go#web_server). So maybe we could be less strict about it?